bridges/snowbridge/pallets/system/src/lib.rs

```diff
@@ -663,6 +722,40 @@ pub mod pallet {
 			let secondary_exists = Channels::<T>::contains_key(SECONDARY_GOVERNANCE_CHANNEL);
 			primary_exists && secondary_exists
 		}
+
+		pub(crate) fn do_register_token(
+			location: &Location,
+			metadata: AssetMetadata,
+			pays_fee: PaysFee<T>,
+		) -> Result<(), DispatchError> {
+			let bridge_location = Location::new(2, [GlobalConsensus(T::EthereumNetwork::get())]);
+			let mut location = location.clone();
+			location
+				.reanchor(&bridge_location, &T::UniversalLocation::get())
+				.map_err(|_| Error::<T>::LocationConversionFailed)?;
+
+			// Record the token id or fail if it has already been created
+			let token_id = TokenIdOf::convert_location(&location)
```

Comment:
what if someone pre-registers this `token_id` on Ethereum? this call will succeed (even the underlying bridge command will fail) and we will be actually be bridging to a contract owned by someone else who can then mint bridged assets out of thin air 🥶 

we could make the registration a two-step process:
1. this call (but asset is in "pending-confirmation")
2. subsequent call providing a proof that snowbridge is the owner of `token_id` on Ethereum (on success this makes the asset "ready to use" over the bridge).

Reply:
I've described in https://github.com/paritytech/polkadot-sdk/pull/5546/files#r1745832990 why that scenario isn't possible.

Reply:
> bridging to a contract owned by someone else who can then mint bridged assets

This is not possible as the owner of the PNA will always be our Gateway contract and not the creator.

Reply:
The only way a bridge command to register a token should be produced is from this call. 

The `onlySelf` modifier ensures only the Gateway contract can call this method and the Gateway only dispatches to the method after proof verification:

1. Proof Verification: 
https://github.com/Snowfork/snowbridge/blob/main/contracts/src/Gateway.sol#L172-L174
1. Dispatch to register token
https://github.com/Snowfork/snowbridge/blob/main/contracts/src/Gateway.sol#L237-L240
1. Securing registration to Gateway via `onlySelf`:
https://github.com/Snowfork/snowbridge/blob/main/contracts/src/Gateway.sol#L420

This means the message must have been created on BH via the register extrinsic.

This `token_id` is not a "converted" to an address directly like a sovereign account. It is used to lookup a contract address in a mapping.
https://github.com/Snowfork/snowbridge/blob/c2af15e16b95633b2ad5939e67d4911efafaea01/contracts/src/Assets.sol#L330-L336

The registration message is the only message that can append to that mapping so this means that no one can squat on a token id.

Reply:
> This token_id is not a "converted" to an address directly like a sovereign account. It is used to lookup a contract address in a mapping.
https://github.com/Snowfork/snowbridge/blob/c2af15e16b95633b2ad5939e67d4911efafaea01/contracts/src/Assets.sol#L330-L336
The registration message is the only message that can append to that mapping so this means that no one can squat on a token id.

Good! this was the part I was missing and concerned with. 👍